BuilderDesignPattern/BuilderDesignPattern/Program.cs

```diff
@@ -0,0 +1,212 @@
+                Pizza result = this.pizza;
+                this.Reset();
+                return result;
+            }
+
+        }
+
+        public class WhitePizzaBuilder : IPizzaBuilder
+        {
+            private Pizza pizza = new Pizza();
+
+            public void Reset()
+            {
+                this.pizza = new Pizza();
+            }
+
+            public void BuildDough()
+            {
+                this.pizza.Add("Add white dough");
+            }
+
+            public void BuildSauce()
+            {
+                this.pizza.Add("Add sauce");
+            }
+
+            public void BuildTopping()
+            {
+                this.pizza.Add("Add topping");
+            }
+
+            public void BuildCheese()
+            {
+                this.pizza.Add("Add cheese");
+            }
+
+            public void BuildDoubleCheese()
+            {
+                this.pizza.Add("Add double cheese");
+            }
+
+            public Pizza GetPizza()
+            {
+                Pizza result = this.pizza;
+                this.Reset();
+                return result;
+            }
+        }
+
+        public class Pizza
+        {
+            private List<object> parts = new List<object>();
+
+            public void Add(string part)
+            {
+                this.parts.Add(part);
+            }
+
+            public string ListParts()
+            {
+                string str = "";
+                for (int i = 0; i < this.parts.Count; i++)
+                {
+                    str += this.parts[i] + ",";
+                }
+
+                return "Pizza parts: " + str;
+            }
+        }
+
+        public class Director
+        {
+            private IPizzaBuilder pizzaBuilder;
+
+            public IPizzaBuilder Builder
+            {
+                set
+                {
+                    this.pizzaBuilder = value;
+                }
+            }
+
+            // minimalist pizza
+            public void BuildMinPizza()
+            {
+                this.pizzaBuilder.BuildDough();
+                this.pizzaBuilder.BuildSauce();
+            }
+
+            // fancy pizza with less cheese
+            public void BuildFancyPizza()
+            {
+                this.pizzaBuilder.BuildDough();
+                this.pizzaBuilder.BuildSauce();
+                this.pizzaBuilder.BuildTopping();
+                this.pizzaBuilder.BuildCheese();
+            }
+
+            // fancy pizza with more cheese
+            public void BuildFancyPizzaDoubleCheese()
+            {
+                this.pizzaBuilder.BuildDough();
+                this.pizzaBuilder.BuildSauce();
+                this.pizzaBuilder.BuildTopping();
+                this.pizzaBuilder.BuildDoubleCheese();
+            }
+        }
+
+        static void Main(string[] args)
+        {
+            var director = new Director();
+
+            var wheatPizzaBuilder = new WheatPizzaBuilder();
```

Comment:
It would be better if you used polymorphism to define builders, without creating new builder variable for each one.